cloud-core/src/main/java/cloud/commandframework/CommandTree.java

```diff
@@ -268,157 +271,182 @@ private CommandTree(final @NonNull CommandManager<C> commandManager) {
     ) {
         CommandPermission permission;
         final List<Node<CommandArgument<C, ?>>> children = root.getChildren();
-        if (children.size() == 1 && !(children.get(0).getValue() instanceof StaticArgument)) {
-            // The value has to be a variable
-            final Node<CommandArgument<C, ?>> child = children.get(0);
-            permission = this.isPermitted(commandContext.getSender(), child);
-            if (!commandQueue.isEmpty() && permission != null) {
-                return Pair.of(null, new NoPermissionException(
-                        permission,
-                        commandContext.getSender(),
-                        this.getChain(child)
-                                .stream()
-                                .filter(node -> node.getValue() != null)
-                                .map(Node::getValue)
-                                .collect(Collectors.toList())
-                ));
-            }
-            if (child.getValue() != null) {
-                if (commandQueue.isEmpty()) {
-                    if (child.getValue().hasDefaultValue()) {
-                        commandQueue.add(child.getValue().getDefaultValue());
-                    } else if (!child.getValue().isRequired()) {
-                        if (child.getValue().getOwningCommand() == null) {
-                            /*
-                             * If there are multiple children with different owning commands then it's ambiguous and
-                             * not allowed, therefore we're able to pick any child command, as long as we can find it
-                             */
-                            Node<CommandArgument<C, ?>> node = child;
-                            while (!node.isLeaf()) {
-                                node = node.getChildren().get(0);
-                                if (node.getValue() != null && node.getValue().getOwningCommand() != null) {
-                                    child.getValue().setOwningCommand(node.getValue().getOwningCommand());
+
+        // Check whether it matches any of the static arguments
+        // If so, do not attempt parsing as a dynamic argument
+        boolean matchesLiteral = false;
+        if (!commandQueue.isEmpty()) {
```

Comment:
Can't we return immediately if we know it's matching a literal? That would get rid of the if statement below

Reply:
Good catch, must've left that in there from earlier. I'll fix that.